lib/platform/bitbucket-server/index.ts

```diff
@@ -153,11 +159,60 @@ export async function getJsonFile(
   return JSON.parse(raw);
 }
 
+function getRepoUrl(
```

Comment:
Please do this function extraction first in a separate PR without any functional change. Then use this branch / PR for the  gitUrl` changes only.